src/block/NetherVines.php

```diff
@@ -0,0 +1,188 @@
+	public function setAge(int $age) : self{
+		if($age < 0 || $age > self::MAX_AGE){
+			throw new \InvalidArgumentException("Age must be in range 0-" . self::MAX_AGE);
+		}
+
+		$this->age = $age;
+		return $this;
+	}
+
+	public function isAffectedBySilkTouch() : bool{
+		return true;
+	}
+
+	public function ticksRandomly() : bool{
+		return true;
+	}
+
+	public function canClimb() : bool{
+		return true;
+	}
+
+	abstract protected function getGrowthFace() : int;
+
+	private function getSupportFace() : int{
+		return Facing::opposite($this->getGrowthFace());
+	}
+
+	private function canBeSupportedBy(Block $block) : bool{
+		return $block->getSupportType($this->getSupportFace())->hasCenterSupport() || $block->isSameType($this);
+	}
+
+	public function onNearbyBlockChange() : void{
+		if(!$this->canBeSupportedBy($this->getSide($this->getSupportFace()))){
+			$this->position->getWorld()->useBreakOn($this->position);
+		}
+	}
+
+	private function seekToTop() : NetherVines{
+		$face = $this->getGrowthFace();
+		$top = $this;
+		while(($next = $top->getSide($face)) instanceof NetherVines && $next->isSameType($this)){
+			$top = $next;
+		}
+		return $top;
+	}
+
+	public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Block $blockClicked, int $face, Vector3 $clickVector, ?Player $player = null) : bool{
+		if(!$this->canBeSupportedBy($blockReplace->getSide($this->getSupportFace()))){
+			return false;
+		}
+		$this->age = mt_rand(0, self::MAX_AGE - 1);
+		return parent::place($tx, $item, $blockReplace, $blockClicked, $face, $clickVector, $player);
+	}
+
+	public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player $player = null, array &$returnedItems = []) : bool{
+		if($item instanceof Fertilizer){
+			if($this->seekToTop()->getSide($this->getGrowthFace())->canBeReplaced()){
+				if($this->grow(mt_rand(1, 5))){
+					$item->pop();
+				}
+			}
+			return true;
+		}
+		return false;
+	}
+
+	public function onRandomTick() : void{
+		if(mt_rand(0, 100) < 10 && $this->age < self::MAX_AGE){
+			if($this->getSide($this->getGrowthFace())->canBeReplaced()){
+				$this->grow();
+			}
+		}
+	}
+
+	private function grow(int $growthAmount = 1) : bool{
+		$top = $this->seekToTop();
+		$world = $this->position->getWorld();
+		$growthFace = $this->getGrowthFace();
+		$supportFace = $this->getSupportFace();
+		$grew = false;
+
+		for($i = 1; $i <= $growthAmount && $top->getSide($growthFace, $i)->canBeReplaced(); $i++){
+			$pos = $top->getPosition()->getSide($growthFace, $i);
+			if(!$world->isInWorld($pos->getFloorX(), $pos->getFloorY(), $pos->getFloorZ())){
+				break;
+			}
+
+			$block = $world->getBlock($pos->getSide($supportFace));
+
+			if($block instanceof NetherVines){
+				$ev = new BlockGrowEvent($block, (clone $block)->setAge(min($block->getAge() + 1, self::MAX_AGE)));
+				$ev->call();
+
+				if($ev->isCancelled()){
+					break;
+				}
+				$world->setBlock($pos, $ev->getNewState());
+				$grew = true;
+			}
```

Comment:
I can't make head or tail of this code. It doesn't appear to be actually growing the vines at all, just updating the age of the existing vine blocks if they exist.

Reply:
In pm3 growth was normal. Now it's broken

Reply:
In pm3 growth was normal. Now it's broken

Reply:
This code existed in neither PM3 nor PM4. Stop hijacking the conversation.

Reply:
This code is cloning the block, increase his age and setting it on a new position. I don't use Vanilla Blocks here because it may be weeping or twisting vine.

Should i use VanillaBlocks + match() here?

Reply:
I wrote this to the fact that the mechanics of growth, which ipad54 showed normal for the first time. Honestly, I did not see any strange growth behavior.

Reply:
I wrote this to the fact that the mechanics of growth, which ipad54 showed normal for the first time. Honestly, I did not see any strange growth behavior.

Reply:
> This code is cloning the block, increase his age and setting it on a new position.

BlockGrowEvent doesn't show that. It shows that the vine block is changing its own age.

This code is very confusing and needs improvement.